### ISSUE TITLE
Code for creating full-width floating window no longer works

### ISSUE THREAD

**Issue by resolritter** [neovim/neovim#11012]
## `git bisect` results

```
d11a146516557fe4298f7af45c789346cc41fbd8 is the first bad commit
commit d11a146516557fe4298f7af45c789346cc41fbd8
Author: Björn Linse <bjorn.linse@gmail.com>
Date:   Mon Jun 10 15:42:09 2019 +0200

    window: allow resize wincmds for floats

:040000 040000 19acbf372846be1889f046f1e8065cd04624d653 7090fda03bfca3b05d0cbef792bdfca0c7372042 M      src
:040000 040000 272fedfb44617df9700b430e93196f6d1614a479 e16c85836de08ee69e45d90d535088b53ec78d0d M      test
```

---

- `nvim --version`:

```
NVIM v0.4.0-2053-g7652904f7
Build type: RelWithDebInfo
LuaJIT 2.0.5
Compilation: /usr/bin/gcc-5 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -DMIN_LOG_LEVEL=3 -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/home/travis/build/neovim/bot-ci/build/neovim/build/config -I/home/travis/build/neovim/bot-ci/build/neovim/src -I/home/travis/build/neovim/bot-ci/build/neovim/.deps/usr/include -I/usr/include -I/home/travis/build/neovim/bot-ci/build/neovim/build/src/nvim/auto -I/home/travis/build/neovim/bot-ci/build/neovim/build/include
Compiled by travis@travis-job-f5953c50-5a72-4e14-867f-02f663d6e928
```
- Operating system/version: 

```
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 19.04
Release:        19.04
Codename:       disco
```

- Terminal name/version:
`
kitty 0.14.4`

- `$TERM`:
`xterm-kitty`

### Steps to reproduce using `nvim -u NORC`

```
  let buf = nvim_create_buf(v:false, v:true)
  call setbufvar(buf, '&signcolumn', 'no')

  let height = &lines - 3
  let width = &columns
  let col = float2nr((&columns - width) / 2)

  let opts = {
        \ 'relative': 'editor',
        \ 'row': 1,
        \ 'col': col,
        \ 'width': width,
        \ 'height': height
        \ }

  let win = nvim_open_win(buf, v:true, opts)
```

### Actual behaviour

Floating window takes the whole screen

### Expected behaviour

Floating window only takes a portion of the screen on the left side of the terminal

**Comment by bfredl** [comment]
Cannot reproduce with 3855204f58. works as expected both with `nvim -u NONE` and my config.

**Comment by reaysawa** [comment]
I'll give more detail to my use-case. I'm using the code posted [here](https://github.com/junegunn/fzf.vim/issues/664#issuecomment-476438294) to make FZF fill the whole screen in a floating window.

I hadn't looked at the diff before because I was re-configuring my setup in a new machine at work when this behavior came up, so I apologize; my hint is that it has something to do with the 9999 being changed to Columns. Here's how it looks like:

![image](https://user-images.githubusercontent.com/17429390/64916902-11ae2080-d75f-11e9-8c95-fa698cb95b58.png)

I thought of using a floating window because I didn't want the newly-created buffer to interrupt me in the middle of editing - the suggested `enew` throws "E37: No write since last change" on modified buffers, as expected, and that bothered me. However, nowadays I realize I can just use `tabnew` to not run in this issue and have changed the command to that, effectively solving this use-case specifically.

```diff
-let g:fzf_layout = { 'window': 'call FloatingFZF()' }
+let g:fzf_layout = { 'window': 'tabedit' }
```

I can't think of uses I'd have outside of that, but can help to figure out why it stopped working at that specific commit if you're interested in pursuing this further. As far as I'm concerned, I'm fine with how I had to change the code. Thank you in advance!

**Comment by bfredl** [comment]
>  my impression is that it has something to do with the 9999 being changed to Columns.

This change is only inside the implementation of resize wincmds. You have to show then where and how you are using a resize command.

**Comment by reaysawa** [comment]
I did some experimenting prior to posting the second time here; for instance, creating the floating window prior to calling `fzf#run` and setting an empty string to the 'window' option of g:fzf_layout, among other things. However, the actual running and setting up of the buffer only happens when you call into FZF (i.e. you build every option first and then pass them all at once to FZF, only gaining control back when the process closes), so it's hard to follow from the outside. Alternatively, I tried also to see where this 'window' option is actually executed inside of the code and IIRC it gets parsed into a command-line flag.

Moral of the story is I could not figure out at which point things go awry is from my initial efforts. If I can manage some more time for figuring this out, I'll post it here.

---

`Files` at the end of the script below is an FZF command, so the code will only work if FZF is loaded. I was able to reproduce this problem on two different OS (Arch and Ubuntu).

```
let $FZF_DEFAULT_OPTS='--layout=reverse'
let g:fzf_layout = { 'window': 'call FloatingFZF()' }

function! FloatingFZF()
  let buf = nvim_create_buf(v:false, v:true)
  call setbufvar(buf, '&signcolumn', 'no')

  let height = &lines - 3
  let width = &columns
  let col = float2nr((&columns - width) / 2)

  let opts = {
        \ 'relative': 'editor',
        \ 'row': 1,
        \ 'col': col,
        \ 'width': width,
        \ 'height': height
        \ }

  call nvim_open_win(buf, v:true, opts)
endfunction

Files
```

**Comment by resolritter** [comment]
Not a bug. I had forgotten I had a weird setup tied to `TermOpen`, which ended up being the culprit here. Strangely the resize in that trigger was being "overridden" before that commit I posted in the OP. It seems like I was relying on a bug.

My apologies and thanks for your great work @bfredl. Closing!
